docs/source/data-access-api/calibration-files.rst

```diff
@@ -0,0 +1,146 @@
+
+This document lays out the requirements for calibration files in the SDC and the expected file naming
+convention.
+
+File Naming Convention
+----------------------
+
+File overview
+^^^^^^^^^^^^^^
+
+In this document, I describe two main kinds of calibration files:
+
+#. Universal calibration files
+#. Time window calibration files
+
+**Universal calibration files** are files which are expected to change rarely. Generally, there should be one file
+which covers all times.
+
+**Time window calibration files** are files which are expected to change more frequently, and are only valid for a certain
+time range.
+
+The main distinction between these types of files is the way they are applied to reprocessing. Universal calibration files would generally
+use the most recent version for all time ranges and reprocessing. Time window calibration files should be defined only for a specific time range,
+and reprocessing should usually use the same file as the first processing.
+
+These two types of files are mostly used for creating the filename and processing conventions, so if your calibration doesn't neatly fit into
+either category, that's okay. The main thing is that all files should try and follow the conventions in this document.
+
+Specific examples
+^^^^^^^^^^^^^^^^^
+
+As an example, we will be looking at the Mag calibration files. These are supplemental files provided by the instrument team for use in
+L1 and L2 processing.
+
+The first file (L1B rotation matrices) is expected to change rarely,
+with the most recent version being used for all data. This file contains rotation calibration matrices from
+the boom to the instrument. These files would only change if there was a physical change to the boom or if there is
+some more accurate information available, so they are universal calibration files.
+
+The second file contains matrices which are applied to create the L2 and L2pre data. They are updated
+approximately monthly, and contain a start and end time for validity. During reprocessing, the data file with the most
+recent version and a time range that includes the data will be used. Since these are files which specifically cover a time range,
+they are time window calibration files.
+
+These two cases cover the two main types of calibration files. They both will be uploaded to the SDC
+with the API, and the correct file to use will be determined by the filename at processing time.
+
+File name format
+^^^^^^^^^^^^^^^^
+
+The file name for each file should be the following format::
+
+    <instrument>_<calibration_name>_<start_date>_<end_date>_<version>.<extension>
+
+Since these names are only used in instrument processing, they are flexible and can be adapted to
+accommodate different needs. This is more of a rule of thumb than a strict requirement.
+
+* ``<instrument>``: The instrument name associated with the calibration file
+* ``<calibration_name>``: A descriptive name for the calibration file which distinguishes between
```

Comment:
I wonder if we should _require_ that the word `calibration` (or perhaps `cal` for short) be in the `calibration_name` so that it can be easily distinguished / searched on from our other data?

Reply:
I like the idea too. If not in the name, is there a database field where file type could be tracked to easily query these files?

Reply:
Yeah, it almost certainly would fall under the indexer lambda, which currently adds a line in a database for science files to track them. Presumably we would track these the same way. Also, they will be stored in a different location in S3, so at least from the SDC side it will be easy to distinguish them from everything else. But, if you just have a file in isolation, for example in an archive, it wouldn't be clear... maybe we can just require pre-approval from an SDC member of the names to make sure they're fully descriptive? Thoughts @bourque ?

Reply:
I was thinking about this from the perspective of a user who just has a directory with a whole bunch of various files from an instrument in it. I reckon it would be nice for this hypothetical user to be able to do something like `ls *cal*` or something along those lines. I'm not sure how important or relevant this would be in practice though.

Reply:
🤔 Interesting idea. I do think that should be up to the instrument teams, if they put "cal" in their names to distinguish them then that's valid... but I'm very on the fence about it, I'm happy to call me outvoted if you and Veronica think it's useful!

Reply:
How is data delivered to users? If it's all placed in one directory, I think there is some value to distinguishing files via their names. When delivering ocean acoustic data to the public in my last job, calibration and ancillary data were organized in their own directories since the user community needed this level of granularity. Do we know what the user community needs here? Perhaps the instrument teams can provide some insight?

Reply:
That's a good way of looking at it, I think. We'll deliver all files through the API so we have a lot of control over what data delivery looks like to end users. In the backend, they will also be separated in different directories. I think if we followed the existing patterns, users would find ancillary files by making a call like `imap-data-access --instrument mag --ancillary` which would only return ancillary files